sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala

```diff
@@ -597,27 +601,28 @@ abstract class UpgradesSpec(val suffix: String)
   override lazy val devMode = true;
   override val cantonFixtureDebugMode = CantonFixtureDebugKeepTmpFiles
 
-  protected def loadPackageIdAndBS(path: String): Future[(PackageId, ByteString)] = {
+  protected def loadPackageIdAndBS(path: String): (PackageId, ByteString) = {
     val dar = DarReader.assertReadArchiveFromFile(new File(BazelRunfiles.rlocation(path)))
     assert(dar != null, s"Unable to load test package resource '$path'")
 
-    val testPackage = Future {
+    val testPackage = {
       val in = new FileInputStream(new File(BazelRunfiles.rlocation(path)))
       assert(in != null, s"Unable to load test package resource '$path'")
       in
     }
-    val bytes = testPackage.map(ByteString.readFrom)
-    bytes.onComplete(_ => testPackage.map(_.close()))
-    bytes.map((dar.main.pkgId, _))
+    val bytes = ByteString.readFrom(testPackage)
+    testPackage.close()
+    dar.main.pkgId -> bytes
```

Comment:
I'm not convinced that this will always close the underlying file object - particularly if the assert (or anything else on line 609) throws.

Maybe it would be better to write something like:
```
import scala.util.Using

val bytes = Using(new FileInputStream(new File(BazelRunfiles.rlocation(path)))) { in =>
   ByteString.readFrom(in)
}
```